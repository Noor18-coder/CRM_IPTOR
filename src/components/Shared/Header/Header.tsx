import React from 'react';
import { Dispatch } from "redux";
import { useSelector, useDispatch } from "react-redux";

import ImageConfig from '../../../config/ImageConfig';
import { getIntialsFromFullName } from '../../../helpers/utilities/lib';

import { UserItem, CompanyInfoItem } from "../../../helpers/Api/models";
import { logOut } from "../../../store/Auth/Actions";
import { AppState } from "../../../store/store";
import { OverlayTrigger, Popover, Image} from 'react-bootstrap'

const Header = () => {

    const state: UserItem = useSelector((state: AppState) => state.auth.user);
    const dispatch: Dispatch<any> = useDispatch();

    const userProfileActions = () => {
        dispatch(logOut());
    }
    const getCompanyName = () => {
      
        if (state.currentEnvironment) {
            const data = state.currentEnvironment.find((obj: CompanyInfoItem) => { return obj.companyCode == state.selectedCompany })
            const name = data && data.name ? data.name : ''
            return name
        }
      
          return ''  
    }
    const popover = (
        <Popover id="popover-basic" className="tool-tip">
          <Popover.Content>
            { state.description}
          </Popover.Content>
          <Popover.Content>
            <a onClick={userProfileActions} role="button">
            Logout <Image className="logout-image" height="15" src={ImageConfig.LOGOUT_ICON} alt="Iptor" title="Iptor"/>              
            </a>
          </Popover.Content>
        </Popover>
      );

    return (
        <>
            <nav className={"site-navbar navbar navbar-default bg-black"} role="navigation">

                <div className={"navbar-container container-fluid"}>
                    <div className={"navbar-header"}>
                        <div className={"navbar-brand navbar-brand-center navbar-sec-logo"}>
                            <img className={"navbar-brand-logo"} src={ImageConfig.IPTOR_ICON} title={"iptor"} />
                        </div>
                        <div className={"navbar-brand navbar-brand-center navbar-sec-flag"}>
                            <span className={"navbar-brand-text hidden-xs-down"}> {state.currentEnvironment ?  getCompanyName() : ''}</span>
                        </div>
                    </div>


                    <ul className={"nav navbar-toolbar navbar-right navbar-toolbar-right"}>

                        <li className={"nav-item"}>
                            <a className={"nav-link navbar-avatar"} href="#" aria-expanded="false" data-animation="scale-up" role="button">
                                <span className={"avatar avatar-online"}>
                                    <img src={ImageConfig.NAV_DASHBOARD_ICON} alt="..." />
                                    <i>Dashboard</i>
                                </span>
                            </a>
                        </li>

                        <li className={"nav-item active"}>
                            <a className={"nav-link navbar-avatar"} href="#" aria-expanded="false" data-animation="scale-up" role="button">
                                <span className={"avatar avatar-online"}>
                                    <img src={ImageConfig.NAV_OPPTY_ICON} alt="..." />
                                    <i>Opportunities</i>
                                </span>
                            </a>
                        </li>


                        <li className="nav-item">
                            <a className={"nav-link navbar-avatar"} href="#" aria-expanded="false" data-animation="scale-up" role="button">
                                <span className={"avatar avatar-online"}>
                                    <img src={ImageConfig.NAV_CUSTOMER_ICON} alt="..." />
                                    <i>Customers</i>
                                </span>
                            </a>
                        </li>


                        <li className={"nav-item"}>
                            <a className={"nav-link navbar-avatar"} href="#" aria-expanded="false" data-animation="scale-up" role="button">
                                <span className={"avatar avatar-online"}>
                                    <img src={ImageConfig.NAV_REPORTS_ICON} alt="..." />
                                    <i>Reports</i>
                                </span>
                            </a>
                        </li>


                        <li className={"nav-item"}>
                            <a className={"nav-link navbar-avatar"} aria-expanded={"false"} data-animation={"scale-up"} role="button">
                                <span className={"avatar avatar-online"}>
                                    <img src={ImageConfig.NAV_NOTIFICATION_ICON} alt={"..."} />
                                    <i>Notifications</i>
                                    <span className={"notifications-badge"}></span>
                                </span>
                            </a>
                        </li>
                    </ul>

                    <ul className={"nav navbar-toolbar"}>
                        <li className={"nav-item no-p"}>
                            <a className={"nav-link navbar-avatar"} href="#" aria-expanded="false" data-animation="scale-up" role="button">
                                <span className={"avatar avatar-online"}>
                                    <img src={ImageConfig.NAV_SEARCH_ICON} alt="..." />
                                    <i></i>
                                </span>
                            </a>
                        </li>
                        <OverlayTrigger trigger="click" placement="bottom" overlay={popover}>
                            <button className="logout"> {state.description && getIntialsFromFullName(state.description) }</button>
                        </OverlayTrigger>
                    </ul>
                </div>
            </nav>
        </>

    );
}

export default Header;