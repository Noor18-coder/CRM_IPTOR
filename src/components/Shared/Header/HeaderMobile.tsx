import React from 'react';
import { Image } from 'react-bootstrap';
import { Dispatch } from 'redux';
import { useDispatch, useSelector } from 'react-redux';
import { UserItem, CompanyInfoItem } from '../../../helpers/Api/models';
import { logOut } from '../../../store/Auth/Actions';
import ImageConfig from '../../../config/ImageConfig';
import { AppState } from '../../../store/store';

const HeaderMobile = (): JSX.Element => {
  const state: UserItem = useSelector((appState: AppState) => appState.auth.user);
  const dispatch: Dispatch<any> = useDispatch();
  const toggleLogoutDrawer = () => {
    document.body.style.overflowY = 'hidden';

    const mLogout = document.getElementById('mob-logout') as HTMLElement;

    mLogout.style.display = 'block';
  };
  const HideLogout = () => {
    const mLogout = document.getElementById('mob-logout') as HTMLElement;

    mLogout.style.display = 'none';
  };
  const getCompanyName = () => {
    if (state.currentEnvironment) {
      const data = state.currentEnvironment.find((obj: CompanyInfoItem) => {
        return obj.companyCode === state.selectedCompany;
      });
      return data && data.name ? data.name : '';
    }

    return '';
  };

  const logout = () => {
    dispatch(logOut());
  };
  return (
    <>
      <div className="dshbrd-sticky-header">
        <div className="dshbrd-logo-row">
          <div className="row">
            <div className="col-9 lft-col pr-0">
              <ul className="list-inline">
                <li className="list-inline-item">
                  <Image src={ImageConfig.MOB_IPTOR_LOGO} />
                </li>
                <li className="list-inline-item">{state.currentEnvironment ? getCompanyName() : ''}</li>
              </ul>
            </div>
            <div className="col-3 pl-0 rgt-col text-right">
              <ul className="list-inline">
                {/*
                // Commented Non- Functioning Buttons
                 <li className="list-inline-item">
                  <Image src={ImageConfig.MOB_HEADER_SEARCH} />
                </li> */}
                <li className="list-inline-item">
                  <Image src={ImageConfig.HEADER_USER_ICON} className="dshbrd-mob-usericon" onClick={toggleLogoutDrawer} />
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>

      <div id="mob-logout" role="presentation" onKeyDown={HideLogout} onClick={HideLogout} className="mobview-user-prof-container">
        <div className="panel-container">
          <ul className="user-settings-list">
            <li className="username">{state.description}</li>
            <li onClick={logout} role="presentation" onKeyDown={logout} className="mob-logout">
              Logout
            </li>
          </ul>
        </div>
      </div>
    </>
  );
};
export default HeaderMobile;
