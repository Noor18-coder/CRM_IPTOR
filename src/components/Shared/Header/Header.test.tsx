import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Header from './Header';
import StoreMock from '../../../mocks/Store.mock';

const store = StoreMock.createInitialStore();

describe('[Shared] Header', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header page={0} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
