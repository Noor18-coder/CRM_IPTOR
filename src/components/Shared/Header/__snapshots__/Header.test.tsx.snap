// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Shared] Header should renders correctly 1`] = `
<nav
  className="site-navbar navbar navbar-default bg-black"
  role="navigation"
>
  <div
    className="navbar-container container-fluid"
  >
    <div
      className="navbar-header"
    >
      <div
        className="navbar-brand navbar-brand-center navbar-sec-logo"
      >
        <img
          alt="..."
          className="navbar-brand-logo"
          src="iptor_icon.svg"
          title="iptor"
        />
      </div>
      <div
        className="navbar-brand navbar-brand-center navbar-sec-flag"
      >
        <span
          className="navbar-brand-text hidden-xs-down"
        >
           
          
        </span>
      </div>
    </div>
    <ul
      className="nav navbar-toolbar navbar-right navbar-toolbar-right"
    >
      <li
        className="nav-item active"
      >
        <a
          aria-expanded="false"
          className="nav-link navbar-avatar"
          data-animation="scale-up"
          href="/"
          onClick={[Function]}
          role="button"
        >
          <span
            className="avatar avatar-online"
          >
            <img
              alt="..."
              src="dashboard-selected-icon.svg"
            />
            <i>
              Dashboard
            </i>
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-expanded="false"
          className="nav-link navbar-avatar"
          data-animation="scale-up"
          href="/opportunities"
          onClick={[Function]}
          role="button"
        >
          <span
            className="avatar avatar-online"
          >
            <img
              alt="..."
              src="nav-oppty-icon.svg"
            />
            <i>
              Opportunities
            </i>
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-expanded="false"
          className="nav-link navbar-avatar"
          data-animation="scale-up"
          href="/customers"
          role="button"
        >
          <span
            className="avatar avatar-online"
          >
            <img
              alt="..."
              src="nav-customers-icon.svg"
            />
            <i>
              Customers
            </i>
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-expanded="false"
          className="nav-link navbar-avatar"
          data-animation="scale-up"
          href="/reports"
          role="button"
        >
          <span
            className="avatar avatar-online"
          >
            <img
              alt="..."
              src="nav-report-icon.svg"
            />
            <i>
              Reports
            </i>
          </span>
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-expanded="false"
          className="nav-link navbar-avatar header-notification"
          data-animation="scale-up"
          href="/notifications"
          role="button"
        >
          <span
            className="avatar avatar-online"
          >
            <img
              alt="..."
              src="nav-notifications-icon.svg"
            />
            <i>
              Notifications
            </i>
          </span>
        </a>
      </li>
    </ul>
    <ul
      className="nav navbar-toolbar"
    >
      <button
        className="logout"
        onClick={[Function]}
        type="button"
      >
        
      </button>
    </ul>
  </div>
</nav>
`;
